src/routes/NoteCreateRoute.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { Redirect } from 'react-router';
+import { FormattedMessage } from 'react-intl';
+
+import { NoteCreatePage } from '@folio/stripes/smart-components';
+
+import {
+  DOMAIN_NAME,
+  APP_ICON_NAME,
+} from '../constants';
+import { formatNoteReferrerEntityData } from '../utils';
+
+const NoteCreateRoute = ({
+  location,
+  history,
+}) => {
+  return location.state
+    ? (
+      <NoteCreatePage
+        referredEntityData={formatNoteReferrerEntityData(location.state)}
+        entityTypeTranslationKeys={{ request: 'ui-requests.notes.entityType.request' }}
+        paneHeaderAppIcon={APP_ICON_NAME}
+        paneTitle={<FormattedMessage id="ui-request.notes.newStaffNote" />}
+        domain={DOMAIN_NAME}
+        navigateBack={history.goBack}
```

Comment:
Could you please clarify how this callback eventually used? I have concern over the case when the user opened the page on the route (by URL in the new tab let's say) from where then the user is doing the navigation back without previous history. In this case, the user will be redirected out of the folio app.

Reply:
It works like a simple browser back. In case that you described, user will redirect to ``/request``, because `location.state` should be empty. Also this component was implemented by identity the same one in `ui-eholdings` module - [link](https://github.com/folio-org/ui-eholdings/blob/master/src/routes/note-create.js)

Reply:
Sounds good.